### DISCUSSION TITLE
Login with Microsoft on Bedrock

### DISCUSSION THREAD

**Discussion by mabalIRL** [PrismarineJS/bedrock-protocol#499] [Q&A]
How do I login to my Microsoft account on Bedrock? I have done that on Java but I can't find any code that allows you to specify your login information without having to login again every time the code is ran.

**Comment by extremeheat** [comment]
If you use the client example code in the readme, you sign-in the same way as in mineflayer/node-minecraft-protocol (the same auth library is used).

Once you sign in, your credentials should be cached to your .minecraft folder (~/.minecraft/nmp-cache) so you don't have to sign in. If this is not the case you need to check that the ~/.minecraft folder is writable by the current process.

**Reply by mabalIRL** [reply]
I don't think you understood my question right. I was asking how I could use the function "this.bot = mineflayer.createBot({
            "username": this.username,
            "password": this.password," used in mineflayer/node-minecraft-protocol on bedrock-protocol so I can hardcode my login information without lokens or web login.
Thanks in advance!

**Reply by extremeheat** [reply]
You can try passing the password field as prismarine-auth will accept it, but we do not formally support it here as it's not a reliable form of authentication.

**Comment by mabalIRL** [comment]
I didn't get the password field to work, is there an example for that anywhere or an alternative way? I also wanted to ask if it was possible to use multiple accounts with one js instance. (Aka I just want to have a json file with my logins like mineflayer but for bedrock)
